src/dataset/generic_dataset.py

```diff
@@ -1,8 +1,515 @@
+        img, anns, img_info = self._pre_process_input(frame_data_token)
+
+        height, width = img.shape[0], img.shape[1]
+        c = np.array([img.shape[1] / 2., img.shape[0] / 2.], dtype=np.float32)
+        s = max(img.shape[0], img.shape[1]) * 1.0 if not self.opt.not_max_crop \
+            else np.array([img.shape[1], img.shape[0]], np.float32)
+        aug_s, rot, flipped = 1, 0, 0
+
+        if self.subset == 'train':
+            c, aug_s, rot = self._get_aug_param(c, s, width, height)
+            s = s * aug_s
+            if np.random.random() < self.opt.flip:
+                flipped = 1
+                img = img[:, ::-1, :]
+                anns = self._flip_anns(anns, width)
+
+        trans_input = dataset_utils.get_affine_transform(c, s, rot,
+                                                         [self.opt.input_res.width, self.opt.input_res.height])
+        trans_output = dataset_utils.get_affine_transform(c, s, rot,
+                                                          [self.opt.output_res.width, self.opt.output_res.height])
+        inp = self._get_input(img, trans_input)
+        ret = {'image': inp}
+        gt_det = {'bboxes': [], 'scores': [], 'clses': [], 'cts': []}
+
+        pre_cts, track_ids = None, None
+        if self.opt.tracking:
+            pre_image, pre_anns, frame_dist = self._load_pre_data(video_idx, frame_idx)
+            if flipped:
+                pre_image = pre_image[:, ::-1, :].copy()
+                pre_anns = self._flip_anns(pre_anns, width)
+            if self.opt.same_aug_pre and frame_dist != 0:
+                trans_input_pre = trans_input
+                trans_output_pre = trans_output
+            else:
+                c_pre, aug_s_pre, _ = self._get_aug_param(c, s, width, height, disturb=True)
+                s_pre = s * aug_s_pre
+                trans_input_pre = dataset_utils.get_affine_transform(c_pre, s_pre, rot, [self.opt.input_res.width,
+                                                                                         self.opt.input_res.height])
+                trans_output_pre = dataset_utils.get_affine_transform(c_pre, s_pre, rot, [self.opt.output_res.width,
+                                                                                          self.opt.output_res.height])
+            pre_img = self._get_input(pre_image, trans_input_pre)
+            pre_hm, pre_cts, track_ids = self._get_pre_dets(pre_anns, trans_input_pre, trans_output_pre)
+            ret['pre_img'] = pre_img
+            if self.opt.pre_hm:
+                ret['pre_hm'] = pre_hm
+
+        self._init_ret(ret, gt_det)
+        calib = self._get_calib(img_info, width, height)
+
+        num_objs = min(len(anns), self.max_objs)
+        for k in range(num_objs):
+            ann = anns[k]
+            cls_id = int(ann['category_id'])
+            if cls_id > self.num_classes or cls_id <= -999:
```

Comment:
`-999` could be a named constant, such as `INVALID_CLS_ID`, if this value has some sort of special meaning